docs/en/sql-reference/functions/date-time-functions.md

```diff
@@ -2574,12 +2588,456 @@ SELECT
+- `num`: Number of microseconds to add. [(U)Int*](../data-types/int-uint.md), [Float*](../data-types/float.md).
+
+**Returned value**
+- Returns `date_time` plus `num` microseconds. [Date](../data-types/date.md)/[Date32](../data-types/date32.md)/[DateTime](../data-types/datetime.md)/[DateTime64](../data-types/datetime64.md).
+
+**Example**
+
+```sql
+WITH
+    toDateTime('2024-01-01 00:00:00') AS date_time,
+    '2024-01-01 00:00:00' AS date_time_string
+SELECT
+    addMicroseconds(date_time, 1000000) AS add_microseconds_with_date_time,
+    addMicroseconds(date_time_string, 1000000) AS add_microseconds_with_date_time_string
+```
+
+```response
+┌─add_microseconds_with_date_time─┬─add_microseconds_with_date_time_string─┐
+│      2024-01-01 00:00:01.000000 │             2024-01-01 00:00:01.000000 │
+└─────────────────────────────────┴────────────────────────────────────────┘
+```
+
+## addNanoseconds
+
+Adds a specified number of microseconds to a date with time or a string-encoded date with time.
+
+**Syntax**
+
+```sql
+addNanoseconds(date_time, num)
+```
+
+**Parameters**
+
+- `date_time`: Date with time to add specified number of nanoseconds to. [DateTime](../data-types/datetime.md)/[DateTime64](../data-types/datetime64.md), [String](../data-types/string.md).
+- `num`: Number of nanoseconds to add. [(U)Int*](../data-types/int-uint.md), [Float*](../data-types/float.md).
+
+**Returned value**
+- Returns `date_time` plus `num` nanoseconds. [Date](../data-types/date.md)/[Date32](../data-types/date32.md)/[DateTime](../data-types/datetime.md)/[DateTime64](../data-types/datetime64.md).
+
+**Example**
+
+```sql
+WITH
+    toDateTime('2024-01-01 00:00:00') AS date_time,
+    '2024-01-01 00:00:00' AS date_time_string
+SELECT
+    addNanoseconds(date_time, 1000) AS add_nanoseconds_with_date_time,
+    addNanoseconds(date_time_string, 1000) AS add_nanoseconds_with_date_time_string
+```
+
+```response
+┌─add_nanoseconds_with_date_time─┬─add_nanoseconds_with_date_time_string─┐
+│  2024-01-01 00:00:00.000001000 │         2024-01-01 00:00:00.000001000 │
+└────────────────────────────────┴───────────────────────────────────────┘
+```
+
+## addInterval
+
+Adds an interval to another interval or tuple of intervals.
+
+**Syntax**
+
+```sql
+addInterval(interval_1, interval_2)
+```
+
+**Parameters**
+
+- `interval_1`: First interval or interval of tuples. [interval](../data-types/special-data-types/interval.md), [tuple](../data-types/tuple.md)([interval](../data-types/special-data-types/interval.md)).
+- `interval_2`: Second interval to be added. [interval](../data-types/special-data-types/interval.md).
+
+**Returned value**
+- Returns a tuple of intervals. [tuple](../data-types/tuple.md)([interval](../data-types/special-data-types/interval.md)).
+
+:::note
+If the types of the first interval (or the interval in the tuple) and the second interval are the same they will be merged into one interval.
```

Comment:
Should that be `or the intervals in the tuple`?

Reply:
More importantly, a reader won't understand what it means to "merge intervals into one intervals". Can we re-phrase a bit?